### ISSUE TITLE
[idea] use passportjs as auth plugin

### ISSUE THREAD

**Issue by sbechet** [jscas/cas-server#4]
It can be interesting to create a [passportjs](http://passportjs.org/) auth plugin for cas-server.

**Comment by jsumners** [comment]
I agree. In fact, I started writing one a few months ago. But I need to figure out a proper API for supporting alternate authentication mechanisms. My idea is that plugins will get an instance of https://github.com/jscas/cas-server/blob/master/lib/casInterface.js so that they can fully handle everything that is necessary (generate tickets, create user sessions, etc). I just haven't had time to solidify that idea and implement it.

Do you have any ideas?

**Comment by sbechet** [comment]
Hello James,

I tried my own implementation [here](https://github.com/sbechet/loopback-component-cas/) with the following constraints:

1. Use [loopback framework](http://loopback.io/) to implement logic account (acl, role...)
1. Same for Applications access
1. Full share between loopback accessToken and TGT so we can use all framwork logic: CAS is a protocol among others
1. Full separation between Visual Parts and CAS part.

Now we can use [passport.js](http://www.passportjs.org/) using [loopback-component-passport](https://github.com/strongloop/loopback-component-passport)

I implemented SAML 1.1  you can recover the [necessary](https://github.com/sbechet/loopback-component-cas/blob/master/protocolSaml.js) for your own implementation (templates [here](https://github.com/sbechet/loopback-component-cas/blob/master/xmlTemplates/validSTSaml.marko) and [here](https://github.com/sbechet/loopback-component-cas/blob/master/xmlTemplates/invalidSTSaml.marko)).

Thank you for your work.

**Comment by jsumners** [comment]
Awesome. SAML support is definitely something JSCAS is missing. I sort of have a need for it, but haven't been able to determine how to make it work as I need. Your example will certainly be a great help.
